# Exploratory Data Analysis (EDA) Dasar pada Dataset Iris

## Hari Senin

Pada praktik ini dilakukan Exploratory Data Analysis (EDA) sederhana menggunakan Python pada dataset Iris.

Analisis yang dilakukan meliputi:
- Memahami dataset
- Statistik deskriptif
- Mean
- Median
- Modus
- Minimum
- Maksimum
- Standar deviasi
- Kuartil
- `describe()`
- `value_counts()`
- `unique()`
- `nunique()`

## 1. Pengenalan Exploratory Data Analysis (EDA)

Exploratory Data Analysis (EDA) adalah proses untuk memahami dan mengeksplorasi data sebelum melakukan analisis lebih lanjut.

EDA digunakan untuk melihat karakteristik data, menemukan pola, mengetahui persebaran data, dan mendapatkan gambaran awal dari sebuah dataset.

### Tujuan EDA

Tujuan dari EDA adalah:
1. Memahami struktur dataset.
2. Mengetahui karakteristik setiap variabel.
3. Mengetahui nilai statistik dasar.
4. Menemukan pola atau informasi awal dari data.
5. Membantu menentukan analisis yang dapat dilakukan selanjutnya.

### Workflow EDA

Workflow EDA secara sederhana adalah:

Dataset → Memahami Data → Statistik Deskriptif → Visualisasi → Menemukan Pola → Insight

### Jenis Analisis EDA

**1. Univariate**

Analisis yang dilakukan pada satu variabel. Contohnya melihat distribusi panjang sepal menggunakan histogram.

**2. Bivariate**

Analisis yang dilakukan pada dua variabel untuk melihat hubungan antarvariabel.

**3. Multivariate**

Analisis yang menggunakan lebih dari dua variabel untuk melihat hubungan atau pola yang lebih kompleks.

In [ ]:
!pip install scikit-learn

In [ ]:
import pandas as pd
from sklearn.datasets import load_iris

In [ ]:
iris = load_iris(as_frame=True)

df = iris.frame

df.head()

In [ ]:
df = df.rename(columns={
    'sepal length (cm)': 'sepal_length',
    'sepal width (cm)': 'sepal_width',
    'petal length (cm)': 'petal_length',
    'petal width (cm)': 'petal_width'
})

df.head()

In [ ]:
df['species'] = df['target'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})

df.head()

## 2. Memahami Dataset

Dataset Iris merupakan dataset yang berisi data pengukuran bunga Iris.

Variabel yang digunakan:
- `sepal_length` = panjang sepal
- `sepal_width` = lebar sepal
- `petal_length` = panjang petal
- `petal_width` = lebar petal
- `species` = jenis bunga

In [1]:
df.info()

NameError: name 'df' is not defined

In [2]:
import pandas as pd
from sklearn.datasets import load_iris

iris = load_iris(as_frame=True)

df = iris.frame

df = df.rename(columns={
    'sepal length (cm)': 'sepal_length',
    'sepal width (cm)': 'sepal_width',
    'petal length (cm)': 'petal_length',
    'petal width (cm)': 'petal_width'
})

df['species'] = df['target'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})

df.head()

ModuleNotFoundError: No module named 'sklearn'

In [3]:
import pandas as pd
from sklearn.datasets import load_iris

iris = load_iris(as_frame=True)

df = iris.frame

df = df.rename(columns={
    'sepal length (cm)': 'sepal_length',
    'sepal width (cm)': 'sepal_width',
    'petal length (cm)': 'petal_length',
    'petal width (cm)': 'petal_width'
})

df['species'] = df['target'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})

df.head()

ModuleNotFoundError: No module named 'sklearn'

In [4]:
!pip install scikit-learn

  Using cached scikit_learn-1.9.0-cp311-cp311-win_amd64.whl.metadata (11 kB)
  Using cached joblib-1.6.0-py3-none-any.whl.metadata (6.5 kB)
  Using cached narwhals-2.25.0-py3-none-any.whl.metadata (15 kB)
  Using cached cloudpickle-3.1.2-py3-none-any.whl.metadata (7.1 kB)
Using cached scikit_learn-1.9.0-cp311-cp311-win_amd64.whl (8.3 MB)
Using cached joblib-1.6.0-py3-none-any.whl (306 kB)
Using cached cloudpickle-3.1.2-py3-none-any.whl (22 kB)
Using cached narwhals-2.25.0-py3-none-any.whl (467 kB)

   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------------------------- 0/4 [narwhals]
   ---------------------


[notice] A new release of pip is available: 26.2 -> 26.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [5]:
from sklearn.datasets import load_iris

In [6]:
iris = load_iris(as_frame=True)
df = iris.frame

df = df.rename(columns={
    'sepal length (cm)': 'sepal_length',
    'sepal width (cm)': 'sepal_width',
    'petal length (cm)': 'petal_length',
    'petal width (cm)': 'petal_width'
})

df['species'] = df['target'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})

df.head()

,sepal_length,sepal_width,petal_length,petal_width,target,species
0,5.1,3.5,1.4,0.2,0,setosa
1,4.9,3.0,1.4,0.2,0,setosa
2,4.7,3.2,1.3,0.2,0,setosa
3,4.6,3.1,1.5,0.2,0,setosa
4,5.0,3.6,1.4,0.2,0,setosa


In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   target        150 non-null    int64  
 5   species       150 non-null    str    
dtypes: float64(4), int64(1), str(1)
memory usage: 7.2 KB


In [9]:
df.shape

(150, 6)

In [10]:
df.describe()

,sepal_length,sepal_width,petal_length,petal_width,target
count,150.000000,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333,1.000000
std,0.828066,0.435866,1.765298,0.762238,0.819232
min,4.300000,2.000000,1.000000,0.100000,0.000000
25%,5.100000,2.800000,1.600000,0.300000,0.000000
50%,5.800000,3.000000,4.350000,1.300000,1.000000
75%,6.400000,3.300000,5.100000,1.800000,2.000000
max,7.900000,4.400000,6.900000,2.500000,2.000000


In [11]:
df['species'].value_counts()

species
setosa        50
versicolor    50
virginica     50
Name: count, dtype: int64

In [12]:
df['species'].unique()

<StringArray>
['setosa', 'versicolor', 'virginica']
Length: 3, dtype: str

In [13]:
df['species'].nunique()

3

In [14]:
kolom_numerik = [
    'sepal_length',
    'sepal_width',
    'petal_length',
    'petal_width'
]

In [15]:
df[kolom_numerik].mean().round(2)

sepal_length    5.84
sepal_width     3.06
petal_length    3.76
petal_width     1.20
dtype: float64

In [16]:
df[kolom_numerik].median().round(2)

sepal_length    5.80
sepal_width     3.00
petal_length    4.35
petal_width     1.30
dtype: float64

In [17]:
df[kolom_numerik].mode().iloc[0]

sepal_length    5.0
sepal_width     3.0
petal_length    1.4
petal_width     0.2
Name: 0, dtype: float64

In [18]:
df[kolom_numerik].min()

sepal_length    4.3
sepal_width     2.0
petal_length    1.0
petal_width     0.1
dtype: float64

In [19]:
df[kolom_numerik].max()

sepal_length    7.9
sepal_width     4.4
petal_length    6.9
petal_width     2.5
dtype: float64

In [20]:
df[kolom_numerik].std().round(2)

sepal_length    0.83
sepal_width     0.44
petal_length    1.77
petal_width     0.76
dtype: float64

In [21]:
df[kolom_numerik].quantile([0.25, 0.50, 0.75]).round(2)

,sepal_length,sepal_width,petal_length,petal_width
0.25,5.1,2.8,1.60,0.3
0.50,5.8,3.0,4.35,1.3
0.75,6.4,3.3,5.10,1.8


In [22]:
statistical_summary = pd.DataFrame({
    'Mean': df[kolom_numerik].mean(),
    'Median': df[kolom_numerik].median(),
    'Modus': df[kolom_numerik].mode().iloc[0],
    'Minimum': df[kolom_numerik].min(),
    'Maksimum': df[kolom_numerik].max(),
    'Standar Deviasi': df[kolom_numerik].std(),
    'Q1': df[kolom_numerik].quantile(0.25),
    'Q2': df[kolom_numerik].quantile(0.50),
    'Q3': df[kolom_numerik].quantile(0.75)
}).round(2)

statistical_summary

,Mean,Median,Modus,Minimum,Maksimum,Standar Deviasi,Q1,Q2,Q3
sepal_length,5.84,5.80,5.0,4.3,7.9,0.83,5.1,5.80,6.4
sepal_width,3.06,3.00,3.0,2.0,4.4,0.44,2.8,3.00,3.3
petal_length,3.76,4.35,1.4,1.0,6.9,1.77,1.6,4.35,5.1
petal_width,1.20,1.30,0.2,0.1,2.5,0.76,0.3,1.30,1.8


In [23]:
with pd.ExcelWriter('Statistical Summary.xlsx') as writer:
    statistical_summary.to_excel(
        writer,
        sheet_name='Statistical Summary'
    )

    df['species'].value_counts().to_excel(
        writer,
        sheet_name='Species Count'
    )

    df.head(10).to_excel(
        writer,
        sheet_name='Data Sample',
        index=False
    )

print("Statistical Summary.xlsx berhasil dibuat!")

Statistical Summary.xlsx berhasil dibuat!


## Kesimpulan

Berdasarkan hasil Exploratory Data Analysis (EDA) dasar pada dataset Iris, terdapat 150 data dengan tiga jenis bunga, yaitu setosa, versicolor, dan virginica. Masing-masing jenis memiliki 50 data.

Penggunaan `describe()` memberikan gambaran statistik dasar dari variabel numerik. Sementara itu, `value_counts()` digunakan untuk mengetahui jumlah data pada setiap jenis bunga. Fungsi `unique()` digunakan untuk melihat kategori yang tersedia dan `nunique()` digunakan untuk mengetahui jumlah kategori unik.

Selanjutnya dilakukan perhitungan mean, median, modus, minimum, maksimum, standar deviasi, dan kuartil pada variabel numerik. Hasil tersebut memberikan gambaran awal mengenai karakteristik dan persebaran data pada dataset Iris.

Hasil EDA dasar ini dapat digunakan sebagai dasar untuk melakukan analisis Univariate pada tahap berikutnya.